packages/main/src/Tab.ts

```diff
@@ -38,162 +44,150 @@ const DESIGN_DESCRIPTIONS = {
-			type: String,
-		},
+	/**
+	 * Defines the component's design color.
+	 * <br><br>
+	 * The design is applied to:
+	 * <ul>
+	 * <li>the component icon</li>
+	 * <li>the <code>text</code> when the component overflows</li>
+	 * <li>the tab selection line</li>
+	 * </ul>
+	 *
+	 * <br><br>
+	 * Available designs are: <code>"Default"</code>, <code>"Neutral"</code>, <code>"Positive"</code>, <code>"Critical"</code> and <code>"Negative"</code>.
+	 *
+	 * <br><br>
+	 * <b>Note:</b> The design depends on the current theme.
+	 * @type {sap.ui.webc.main.types.SemanticColor}
+	 * @defaultvalue "Default"
+	 * @public
+	 * @name sap.ui.webc.main.Tab.prototype.design
+	 */
+	@property({ type: SemanticColor, defaultValue: SemanticColor.Default })
+	design!: SemanticColor;
 
-		/**
-		 * Defines the component's design color.
-		 * <br><br>
-		 * The design is applied to:
-		 * <ul>
-		 * <li>the component icon</li>
-		 * <li>the <code>text</code> when the component overflows</li>
-		 * <li>the tab selection line</li>
-		 * </ul>
-		 *
-		 * <br><br>
-		 * Available designs are: <code>"Default"</code>, <code>"Neutral"</code>, <code>"Positive"</code>, <code>"Critical"</code> and <code>"Negative"</code>.
-		 *
-		 * <br><br>
-		 * <b>Note:</b> The design depends on the current theme.
-		 * @type {sap.ui.webc.main.types.SemanticColor}
-		 * @defaultvalue "Default"
-		 * @public
-		 */
-		design: {
-			type: SemanticColor,
-			defaultValue: SemanticColor.Default,
-		},
+	/**
+	 * Specifies if the component is selected.
+	 *
+	 * @type {boolean}
+	 * @defaultvalue false
+	 * @public
+	 * @name sap.ui.webc.main.Tab.prototype.selected
+	 */
+	@property({ type: Boolean })
+	selected!: boolean;
 
-		/**
-		 * Specifies if the component is selected.
-		 *
-		 * @type {boolean}
-		 * @defaultvalue false
-		 * @public
-		 */
-		selected: {
-			type: Boolean,
-		},
+	@property({ type: Boolean })
+	_selected!: boolean;
 
-		_selected: {
-			type: Boolean,
-		},
+	@property({ type: Object })
+	_realTab!: Tab;
 
-		_realTab: {
-			type: Object,
+	@property({ type: Boolean })
+	_isTopLevelTab!: boolean;
+
+	/**
+	 * Holds the content associated with this tab.
+	 *
+	 * @type {Node[]}
+	 * @public
+	 * @slot
+	 * @name sap.ui.webc.main.Tab.prototype.default
+	 */
+	@slot({
+		type: Node,
+		"default": true,
+		invalidateOnChildChange: {
+			properties: true,
+			slots: false,
 		},
+	})
+	content!: Array<Node>;
 
-		_isTopLevelTab: {
-			type: Boolean,
+	/**
+	 * Defines hierarchies with nested sub tabs.
+	 * <br><br>
+	 * <b>Note:</b> Use <code>ui5-tab</code> and <code>ui5-tab-separator</code> for the intended design.
+	 *
+	 * @type {sap.ui.webc.main.ITab[]}
+	 * @public
+	 * @slot subTabs
+	 * @name sap.ui.webc.main.Tab.prototype.subTabs
+	 */
+	@slot({
+		type: HTMLElement,
+		individualSlots: true,
+		invalidateOnChildChange: {
+			properties: true,
+			slots: false,
 		},
-	},
-	events: /** @lends sap.ui.webc.main.Tab.prototype */ {
-	},
-};
+	})
+	subTabs!: Array<Tab>
```

Comment:
Wouldn't "ITab" also work, if the ITab is the interface, it should be possible and better to use it here as well.